src/nbl/video/IPhysicalDevice.cpp

```diff
@@ -14,45 +14,505 @@ IPhysicalDevice::IPhysicalDevice(core::smart_refctd_ptr<system::ISystem>&& s, co
+    addGLSLDefineToPool(pool,"NBL_GLSL_LIMIT_MAX_DESCRIPTOR_SET_IMAGES",m_properties.limits.maxDescriptorSetImages);
+    addGLSLDefineToPool(pool,"NBL_GLSL_LIMIT_MAX_DESCRIPTOR_SET_STORAGE_IMAGES",m_properties.limits.maxDescriptorSetStorageImages);
+    addGLSLDefineToPool(pool,"NBL_GLSL_LIMIT_MAX_DESCRIPTOR_SET_INPUT_ATTACHMENTS",m_properties.limits.maxDescriptorSetInputAttachments);
+
+    addGLSLDefineToPool(pool,"NBL_GLSL_LIMIT_MAX_TESSELLATION_GENERATION_LEVEL",m_properties.limits.maxTessellationGenerationLevel);
+    addGLSLDefineToPool(pool,"NBL_GLSL_LIMIT_MAX_TESSELLATION_PATCH_SIZE",m_properties.limits.maxTessellationPatchSize);
+    addGLSLDefineToPool(pool,"NBL_GLSL_LIMIT_MAX_TESSELLATION_CONTROL_PER_VERTEX_INPUT_COMPONENTS",m_properties.limits.maxTessellationControlPerVertexInputComponents);
+    addGLSLDefineToPool(pool,"NBL_GLSL_LIMIT_MAX_TESSELLATION_CONTROL_PER_VERTEX_OUTPUT_COMPONENTS",m_properties.limits.maxTessellationControlPerVertexOutputComponents);
+    addGLSLDefineToPool(pool,"NBL_GLSL_LIMIT_MAX_TESSELLATION_CONTROL_PER_PATCH_OUTPUT_COMPONENTS",m_properties.limits.maxTessellationControlPerPatchOutputComponents);
+    addGLSLDefineToPool(pool,"NBL_GLSL_LIMIT_MAX_TESSELLATION_CONTROL_TOTAL_OUTPUT_COMPONENTS",m_properties.limits.maxTessellationControlTotalOutputComponents);
+    addGLSLDefineToPool(pool,"NBL_GLSL_LIMIT_MAX_TESSELLATION_EVALUATION_INPUT_COMPONENTS",m_properties.limits.maxTessellationEvaluationInputComponents);
+    addGLSLDefineToPool(pool,"NBL_GLSL_LIMIT_MAX_TESSELLATION_EVALUATION_OUTPUT_COMPONENTS",m_properties.limits.maxTessellationEvaluationOutputComponents);
+    addGLSLDefineToPool(pool,"NBL_GLSL_LIMIT_MAX_GEOMETRY_SHADER_INVOCATIONS",m_properties.limits.maxGeometryShaderInvocations);
+    addGLSLDefineToPool(pool,"NBL_GLSL_LIMIT_MAX_GEOMETRY_INPUT_COMPONENTS",m_properties.limits.maxGeometryInputComponents);
+    addGLSLDefineToPool(pool,"NBL_GLSL_LIMIT_MAX_GEOMETRY_OUTPUT_COMPONENTS",m_properties.limits.maxGeometryOutputComponents);
+    addGLSLDefineToPool(pool,"NBL_GLSL_LIMIT_MAX_GEOMETRY_OUTPUT_VERTICES",m_properties.limits.maxGeometryOutputVertices);
+    addGLSLDefineToPool(pool,"NBL_GLSL_LIMIT_MAX_GEOMETRY_TOTAL_OUTPUT_COMPONENTS",m_properties.limits.maxGeometryTotalOutputComponents);
+    addGLSLDefineToPool(pool,"NBL_GLSL_LIMIT_MAX_FRAGMENT_INPUT_COMPONENTS",m_properties.limits.maxFragmentInputComponents);
+    addGLSLDefineToPool(pool,"NBL_GLSL_LIMIT_MAX_FRAGMENT_OUTPUT_ATTACHMENTS",m_properties.limits.maxFragmentOutputAttachments);
+    addGLSLDefineToPool(pool,"NBL_GLSL_LIMIT_MAX_FRAGMENT_COMBINED_OUTPUT_RESOURCES",m_properties.limits.maxFragmentCombinedOutputResources);
+    addGLSLDefineToPool(pool,"NBL_GLSL_LIMIT_MAX_FRAGMENT_DUAL_SRC_ATTACHMENTS",m_properties.limits.maxFragmentDualSrcAttachments);
+    addGLSLDefineToPool(pool,"NBL_GLSL_LIMIT_MAX_COMPUTE_SHARED_MEMORY_SIZE",m_properties.limits.maxComputeSharedMemorySize);
```

Comment:
need to update some of the builtin GLSL because we used this in a few places

Reply:
maxComputeSharedMemorySize wasn't reported in the old version https://github.com/Devsh-Graphics-Programming/Nabla/blob/master/src/nbl/video/IPhysicalDevice.cpp#L13 unless you mean something else

Reply:
I think @achalpandeyy 's Blit PR used the report, but maybe not for GLSL, just on the CPU side